### PR TITLE
25/04 Proyect update

### PR DESCRIPTION
**Pull request by juanan04**
- Added new File -> EmployeeFactory file
- Added the pagination in Employees list and History (Dates) list
- Updated the .env.example with XDG_CONFIG (to use php artisan tinker) and the INITIAL_ADMIN_PASSWORD

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,3 +64,5 @@ If you discover a security vulnerability within Laravel, please send an e-mail t
 ## License
 
 The Laravel framework is open-sourced software licensed under the [MIT license](https://opensource.org/licenses/MIT).
+
+test juan
```
Comment (bmiecielica) [inline README.md]: please remove the test string again

File: app/Http/Controllers/DatesController.php [hunk 2]
```diff
@@ -13,35 +14,35 @@ class DatesController extends Controller
     public function list() {
         $employeesKochen = Employee::where('gekocht', false)->orderByRaw("SUBSTRING_INDEX(name, ' ', -1)")->get();
         $employeesPraesentieren = Employee::where('praesentiert', false)->orderByRaw("SUBSTRING_INDEX(name, ' ', -1)")->get();
-        $dates = Date::all();
-    
+        $dates = Date::orderBy(DB::raw("STR_TO_DATE(date, '%d.%m.%Y')"), 'desc')->simplePaginate(5);
```
Comment (bmiecielica) [inline app/Http/Controllers/DatesController.php]: Why do we need STR_TO_DATE ?

isn't Date:orderByDesc('date')->simplePaginate(5); also possible?

File: app/Http/Controllers/EmployeesController.php [hunk 3]
```diff
@@ -33,20 +33,24 @@ public function displayImage($fileName)
...
-        });
-    
+        // $employees = Employee::all()->sortBy(function($employee) {
+        //     // Teilen Sie den Namen durch Leerzeichen in Teile auf und geben Sie den letzten Teil (Nachname) zurück
+        //     $lastNames = explode(' ', $employee->name);
+        //     return Str::lower(end($lastNames)); // Stellen Sie sicher, dass Sie Zeichenfolgen in Kleinbuchstaben vergleichen
+        // })->simplePaginate(10);
+
+        // Order by the name of Employee and adding a simplePaginate (max 10 employees for page)
+        $employees = Employee::orderBy('name')->simplePaginate(10);
```
Comment (bmiecielica) [inline app/Http/Controllers/EmployeesController.php]: Was this a requirement? now the names are not sorted by Last name any more

**Review by bmiecielica — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/DatesController.php (hunk 2 above)
Comment (juanan04) [reply app/Http/Controllers/DatesController.php]: I tried but it doesn't order it from the most recent date to the oldest.

For example if i put the date: 01.09.2024 and I've already the date 26.04.2024.It will end up putting the 01.09.2024 date at the end. And this is because the date is saved in the database like a VARCHAR, I guess.

File: README.md (hunk 1 above)
Comment (juanan04) [reply README.md]: It has already been removed.

File: app/Http/Controllers/EmployeesController.php (hunk 3 above)
Comment (juanan04) [reply app/Http/Controllers/EmployeesController.php]: If it is a requirement to order them by the last name, I will try to maintain the previous logic. The reason I commented on it and changed it was because the pagination did not work with sortBy.

File: README.md (hunk 1 above)
Comment (bmiecielica) [reply README.md]: ok, but if i merge this pull request it would be added again, you either have to merge viktorias branch into yours or remove it in your branch

File: app/Http/Controllers/DatesController.php (hunk 2 above)
Comment (bmiecielica) [reply app/Http/Controllers/DatesController.php]: ok then please try to modify the date column to type date and check if it fixes the sorting:
https://laravel.com/docs/11.x/migrations#modifying-columns

File: app/Http/Controllers/EmployeesController.php (hunk 3 above)
Comment (bmiecielica) [reply app/Http/Controllers/EmployeesController.php]: yes, please try to maintain the last name sorting if possible

File: database/migrations/2023_07_07_084116_create_employees_table.php [hunk 4]
```diff
@@ -10,14 +10,16 @@
...
-       
+    {
+
         Schema::create('employees', function (Blueprint $table) {
             $table->id();
             $table->string('name');
             $table->timestamps();
             $table->boolean('praesentiert')->default(false);
             $table->boolean('gekocht')->default(false);
+            $table->boolean('is_available')->default(true);
```
Comment (bmiecielica) [inline database/migrations/2023_07_07_084116_create_employees_table.php]: please create a new migration that modifies the column and revert this change
https://laravel.com/docs/11.x/migrations#modifying-columns

On deployment only new Migrations are executed

File: database/migrations/2023_07_17_111015_create_dates_table.php [hunk 5]
```diff
@@ -10,12 +10,12 @@
...
     public function up(): void
-    {   
-        
+    {
+
         Schema::create('dates', function (Blueprint $table) {
             $table->id();
             $table->timestamps();
-            $table->string('date');
+            $table->date('date');
```
Comment (bmiecielica) [inline database/migrations/2023_07_17_111015_create_dates_table.php]: here also a new migration please

https://laravel.com/docs/11.x/migrations#modifying-columns

File: routes/web.php [hunk 6]
```diff
@@ -25,11 +25,12 @@
 Route::delete('Name/delete/{id}', [EmployeesController::class, 'delete'])->name('name.destroy')->middleware('auth');
 
 Route::get('Verlauf', [DatesController::class, 'list'])->middleware('auth');
-Route::get('/VerlaufKochen/{id1}/{id2}/{id3}', [DatesController::class, 'updatePersonKochen'])->middleware('auth');
-Route::get('/VerlaufPraesentieren/{id1}/{id2}/{id3}', [DatesController::class, 'updatePersonPraesentieren'])->middleware('auth');
+// Route::get('/VerlaufKochen/{id1}/{id2}/{id3}', [DatesController::class, 'updatePersonKochen'])->middleware('auth');
```
Comment (bmiecielica) [inline routes/web.php]: remove pls

**Review by bmiecielica — CHANGES_REQUESTED** [review]

**Review by bmiecielica — APPROVED** [review]
